src/coreclr/inc/check.inl

```diff
@@ -316,10 +316,11 @@ inline CHECK CheckUnderflow(const void *address, UINT64 offset)
     CHECK_OK;
 }
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__wasm__)
 inline CHECK CheckUnderflow(const void *address, SIZE_T offset)
 {
-#if POINTER_BITS == 32
+    // SIZE_T is 32bit on wasm32
+#if !defined(__wasm__) && POINTER_BITS == 32
```

Comment:
Why does this need to be disabled? It seems the check should work on wasm too.

Reply:
This is to avoid 32bit shift of SIZE_T, which is 32bit on wasm.